### ISSUE TITLE
ESLint error

### ISSUE THREAD

**Issue by TkinterinShanghai** [wallabyjs/quokka#764]
### Issue description or question

When eslint is configured for a workspace, ESLint complains with the following message: 
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: test[object Object].
The extension for the file () is non-standard. You should add "parserOptions.extraFileExtensions" to your config.eslint
```

### What I tried

My best guess was using 
```
  "ignorePatterns": [""],
  "overrides": [{ "files": ["**/*.ts"] }],
```

in my `eslintrc.json` file, but that didnt work. Specifying `"ignorePatterns":  ["**/*.ts"] ` did ignore all ts files but not the quokka file, however using `"ignorePatterns":  ["!**/*.ts"]` included both the quokka file and the regular .ts files
### Code editor version

Visual Studio Code v1.65.2

### OS name and version

Windows 10

**Comment by smcenlly** [comment]
From your description, your issue appears to be related to `eslint` and not Quokka. We're unclear exactly how this is related to Quokka but believe you probably need to raise this issue with the `eslint` team.

If you are able to provide us with a repo that includes your `eslint` configuration as well as step-by-step instructions to see your issue using Quokka, we would be happy to check it out for you, but any issue/configuration fix will be related to `eslint`.

**Comment by TkinterinShanghai** [comment]
Knowing the default file name and extension for the quokka file would be helpful. I dont know if it automatically produces either of them but that could fix the issue. If neither of them could be specified, I will head over to the eslint team. Awesome extension in any case

**Comment by smcenlly** [comment]
In the context of creating a new Quokka file, there is no filename provided by our extension. We use the [VS Code API](https://code.visualstudio.com/api/references/vscode-api#workspace), which takes the following parameters:

openTextDocument(options?: {content: [string](), language: [string]()}): [Thenable](https://code.visualstudio.com/api/references/vscode-api#Thenable)<[TextDocument](https://code.visualstudio.com/api/references/vscode-api#TextDocument)>

The `language` is set to `JavaScript` or `TypeScript` depending on the file type you choose to create.

I'm not sure if it helps, but the `uri` property in the result of the newly created document is:

```
{$mid: 1, fsPath: 'Untitled-1', external: 'untitled:Untitled-1', path: 'Untitled-1', scheme: 'untitled'}$mid: 1external: "untitled:Untitled-1"fsPath: "Untitled-1"path: "Untitled-1"scheme: "untitled"[[Prototype]]
```

The `fileName` property is:

```
`Untitled-1`.
```

I expect the `1` increments depending on how many new unsaved documents you create.

**Comment by developer-don** [comment]
I realize this has been closed for some time, however I ran into this exact problem today. 
Adding esnext: true under "env" in module.exports resolved it for me.
Hope it saves someone else time.

Mac M1 Pro on Ventura 13.3.1. 
Node v19.9.0
VsCode v 1.78.0

Some relevant package.json lines...
@typescript-eslint/eslint-plugin: v5.59.5,
@typescript-eslint/parser v5.59.5
eslint: "^8.40.0
typescript ^5.0.4

Everything else about my setup was cookie cutter.
